### PR TITLE
[HiGHS] add explicit Zlib_jll dependency

### PR DESCRIPTION
**Pull request by odow**
x-ref https://github.com/jump-dev/HiGHS.jl/issues/199

TODO:

 - [x] once the build succeeds, I need to check if `Zlib` was correctly found

### CODE REVIEW

File: H/HiGHS/HiGHS/build_tarballs.jl [hunk 1]
```diff
@@ -15,6 +15,7 @@ platforms = expand_cxxstring_abis(platforms)
 
 dependencies = [
     Dependency("CompilerSupportLibraries_jll"),
+    Dependency("Zlib_jll", compat="1"),
```
Comment (odow) [inline H/HiGHS/HiGHS/build_tarballs.jl:18]:
```suggestion
    Dependency("Zlib_jll", compat="~1"),
```

File: H/HiGHS/HiGHSstatic/build_tarballs.jl [hunk 2]
```diff
@@ -19,6 +19,7 @@ dependencies = [
     # is a build-only dependency, in the other cases it's also a runtime one.
     Dependency("CompilerSupportLibraries_jll"; platforms=filter(!Sys.iswindows, platforms)),
     BuildDependency("CompilerSupportLibraries_jll"; platforms=filter(Sys.iswindows, platforms)),
+    Dependency("Zlib_jll", compat="1"),
```
Comment (odow) [inline H/HiGHS/HiGHSstatic/build_tarballs.jl:22]:
```suggestion
    Dependency("Zlib_jll", compat="~1"),
```

File: H/HiGHS/HiGHS/build_tarballs.jl [hunk 3]
```diff
@@ -15,6 +15,7 @@ platforms = expand_cxxstring_abis(platforms)
 
 dependencies = [
     Dependency("CompilerSupportLibraries_jll"),
+    Dependency("Zlib_jll", compat="~1"),
```
Comment (odow) [inline H/HiGHS/HiGHS/build_tarballs.jl:18]:
```suggestion
    Dependency("Zlib_jll"),
```

File: H/HiGHS/HiGHSstatic/build_tarballs.jl [hunk 4]
```diff
@@ -19,6 +19,7 @@ dependencies = [
     # is a build-only dependency, in the other cases it's also a runtime one.
     Dependency("CompilerSupportLibraries_jll"; platforms=filter(!Sys.iswindows, platforms)),
     BuildDependency("CompilerSupportLibraries_jll"; platforms=filter(Sys.iswindows, platforms)),
+    Dependency("Zlib_jll", compat="~1"),
```
Comment (odow) [inline H/HiGHS/HiGHSstatic/build_tarballs.jl:22]:
```suggestion
    Dependency("Zlib_jll"),
```

**Comment by odow** [comment]
Looks like this worked

image

File: H/HiGHS/highs_common.jl [hunk 5]
```diff
@@ -4,7 +4,7 @@ using BinaryBuilder, Pkg
 
 name = "HiGHS"
 
-version = v"1.6.0"
+version = v"1.6.1"
```
Comment (giordano) [inline H/HiGHS/highs_common.jl:7]: The upstream version didn't change though?

Comment (odow) [reply H/HiGHS/highs_common.jl:7]: Yeah upstream hasn't released a new version yet. But I also thought that you couldn't add new dependencies to a `+1` version?

Comment (giordano) [reply H/HiGHS/highs_common.jl:7]: Adding is actually fine, it's removing, or changing compat bounds, that doesn't work

Comment (odow) [inline H/HiGHS/highs_common.jl:7]:
```suggestion
version = v"1.6.0"
```

**Pull request merged**